code/mob/living/critter/flock/flockdrone.dm

```diff
@@ -505,6 +505,12 @@
 			if (wall.on)
 				wall.off()
 	animate_flock_floorrun_end(src)
+	if (istype(src.loc, /turf/simulated/wall/auto/feather))
+		for(var/turf/T in getneighbours(src.loc))
+			if(!flock_is_blocked_turf(T))
+				src.set_loc(T)
+				return
+
```

Comment:
yeah, it makes them basically immune to projectile weapons.

Reply:
Okay, feel like it would be good to let them just stay in walls considering it isn't like that for any other mob, but I guess it's a bit different with it being much easier for Flockdrones. In a way though, you could be just as, or nearly as, tricky with a Flockdrone even if this weren't implemented, since you could go behind and in front of the wall constantly

Reply:
Can they shoot out while in walls? If so then definitely kick them out, we do not want invincible wall drones

Reply:
Think you could do the same with almost anything else you could Floorrun into though, like Flock structures and soon tables/closets if implemented